op-e2e/system/bridge/deposit_test.go

```diff
@@ -127,6 +116,178 @@ func TestDepositTxCreateContract(t *testing.T) {
+	// 6. Ensures the deposit sender's balance is incremented on L2
+
+	op_e2e.InitParallel(t)
+	cfg := e2esys.DefaultSystemConfig(t)
+	cfg.DeployConfig = cfg.DeployConfig.Copy()
+	cfg.DeployConfig.ActivateForkAtGenesis(rollup.Isthmus)
+	delete(cfg.Nodes, "verifier")
+	sys, err := cfg.Start(t)
+	require.NoError(t, err, "Error starting up system")
+
+	l1Client := sys.NodeClient("l1")
+	l2Seq := sys.NodeClient("sequencer")
+
+	// Simple constructor that is prefixed to the actual contract code
+	// Results in the contract code being returned as the code for the new contract
+	deployData := append(deployPrefix, sstoreContract...)
+	signer := types.NewIsthmusSigner(cfg.L2ChainIDBig())
+
+	tx := types.MustSignNewTx(cfg.Secrets.Alice, signer, &types.DynamicFeeTx{
+		ChainID:   cfg.L2ChainIDBig(),
+		Nonce:     0,
+		To:        nil,
+		Value:     big.NewInt(0),
+		Gas:       500000,
+		GasFeeCap: big.NewInt(5000000000),
+		GasTipCap: big.NewInt(2),
+		Data:      deployData,
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	err = l2Seq.SendTransaction(ctx, tx)
+	cancel()
+	require.NoError(t, err, "Failed to send set code transaction")
+
+	ctx, cancel = context.WithTimeout(context.Background(), 15*time.Second)
+	r, err := wait.ForReceipt(ctx, l2Seq, tx.Hash(), 1)
+	cancel()
+	require.NoError(t, err, "Failed to get receipt for set code transaction")
+
+	contractAddr := crypto.CreateAddress(cfg.Secrets.Addresses().Alice, 0)
+	fmt.Println(r.Status, r.ContractAddress, contractAddr)
+
+	// validate codeat
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	code, err := l2Seq.PendingCodeAt(ctx, contractAddr)
+	cancel()
+	require.NoError(t, err, "Failed to get code at contract address")
+	if !bytes.Equal(code, sstoreContract) {
+		t.Fatalf("Code at contract address is incorrect: got %s, want %s", common.Bytes2Hex(code), common.Bytes2Hex(sstoreContract))
+	}
+
+	// set delegation designation for an account on L2 to new contract
+	auth1, err := types.SignSetCode(cfg.Secrets.Bob, types.SetCodeAuthorization{
+		Address: contractAddr,
+		Nonce:   1,
+	})
+
+	require.NoError(t, err, "Failed to sign set code authorization")
+
+	txdata := &types.SetCodeTx{
+		ChainID:   uint256.MustFromBig(cfg.L2ChainIDBig()),
+		Nonce:     0,
+		To:        cfg.Secrets.Addresses().Bob,
+		Gas:       500000,
+		GasFeeCap: uint256.NewInt(5000000000),
+		GasTipCap: uint256.NewInt(2),
+		AuthList:  []types.SetCodeAuthorization{auth1},
+		Data:      []byte{1},
+	}
+
+	tx = types.MustSignNewTx(cfg.Secrets.Bob, signer, txdata)
+
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	err = l2Seq.SendTransaction(ctx, tx)
+	cancel()
+	require.NoError(t, err, "Failed to send set code transaction")
+
+	ctx, cancel = context.WithTimeout(context.Background(), 15*time.Second)
+	_, err = wait.ForReceipt(ctx, l2Seq, tx.Hash(), 1)
+	cancel()
+	require.NoError(t, err, "Failed to get receipt for set code transaction")
```

Comment:
Nit: I wonder if we could increase readability by extracting some of these repeated bits into a helper?

What if we just set up a single context with a deferred cancel at the start of the test, and use it everywhere? I don't feel like all these contexts and cancel functions are doing much for us apart from making the test script a lot longer.